### ISSUE TITLE
[Launcher] Add support for profiles for Nitrox save data and other files

### ISSUE THREAD

**Issue by Measurity** [SubnauticaNitrox/Nitrox#958]
Right now files are saved left and right. It would be nice to have them in one "nitrox data" folder.

Both client and server should store files into same directory if both are on the same PC/machine. Location of the profiles is likely best stored in App Data but if not, let me know.

The API is best implemented in the "NitroxModel" project so that it can be used anywhere.

**Folder structure should be like**

"Nitrox data"
 - "default"
   - "servers" (text file with server list for profile 1; default)
   - "world"
     - "BaseData.nitrox"
     - "PlayerData.nitrox"
     - "WorldData.nitrox"
   - "dev"
     - "settings.json"
 - "Profile 2" (If created, same structure as profile 1; default)
 - ...

**Example API (doesn't need to work like this)**

```cs
var profile = DataProfile.CreateOrLoad("default"); // Ensures folder structure
var serverList = profile.GetServers(); // servers stored in "servers" file or empty list
...
profile.SetServers(servers); // Store servers into its file
profile.GetBaseData(); // Returns an object that's deserialized from base data.
profile.SetBaseData(baseData); // Store base data object to file.

// Delete whole profile folder (could be useful for removing old Nitrox profiles)
profile.Delete();
```

**Comment by killzoms** [comment]
Doesnt this run into the issue of there being specific settings for the server that are pointless on the client? wouldnt it be best to separate the client and server data? Yes I can see where keeping them together is nice however it can cause certain issues to crop up in the long run especially when someone wants to put the server files on a separate drive from their client, Say in the event of someone having a specific drive they want their server to be on(Say for speed purposes) but they don't keep their games/programs on that drive.

**Comment by Measurity** [comment]
We can use the same structure for both client and server with just some files missing. This way we can have the same code for both client and server.

**Comment by Jannify** [comment]
Will look into that after #1075

**Comment by Mathew3000** [comment]
I'll take a look at it. Seems doable :)
